### ISSUE TITLE
If you make breaking change then bump the major version!

### ISSUE THREAD

**Issue by Sauraus** [bflad/chef-docker#338]
you cannot go from 0.37.x to 0.40.x and completely rewrite the cookbook!!

**Comment by scalp42** [comment]
Merging this will help https://github.com/bflad/chef-docker/pull/333

**Comment by someara** [comment]
Merged as c5f2bbb13ac9bad59673db0603557e24e4b88d04

**Comment by someara** [comment]
released as 0.40.2

**Comment by someara** [comment]
Er, sorry, previous comments were for the wrong issue

**Comment by someara** [comment]
@Sauraus @scalp42 

Sorry for the breaking changes on a minor version... this warrants a bit of an explanation.

Initially I meant for the rewrite to become a 1.0.0... however, work stalled and it was sitting in the "develop" branch for a weeks and weeks. People were asking for these bits to get into master sooner than the ill-defined "later". Normally, this would warrant a major version bump, but because of its 0.x.x status, and the fact that Chef doesn't support -alpha and -beta semver semantics, incrementing minor to the next even 10 made the most sense.

From http://semver.org/

4- Major version zero (0.y.z) is for initial development. Anything may change at any time. The public API should not be considered stable

5 - "Version 1.0.0 defines the public API. The way in which the version number is incremented after this release is dependent on this public API and how it changes."

So, basically, the minor version bump was to avoid defining that public API.

-s

**Comment by Sauraus** [comment]
@someara I am familiar with semver however there is room for some critical thinking with standards, because in this case the change clearly blew up in everybody's face and there was no way anyone could have protected themselves from this, with any version constraint except by locking to 1 specific version. Just something to consider for the future. :)

**Comment by someara** [comment]
Indeed. The general plan going forward is a continuous delivery model.






—
